/cwms-data/levels endpoint mixes up office with timezone

Example cURL:
```
curl -X 'GET' \
  'http://localhost:11524/cwms-data/levels?name=AARK.Flow.Inst.0.NWS%20Flood&office=SWT&unit=SI&datum=NAVD88&begin=01Jan1900%2000%3A00&end=01Jan1900%2000%3A00&timezone=UTC' \
  -H 'accept: */*'
```

Server-side stack:
```
Nov 18, 2021 4:24:40 PM cwms.radar.ApiServlet lambda$init$9
WARNING: error on request[-7798373712580180303]: /cwms-data/levels
org.jooq.exception.DataAccessException: SQL [{ ? = call "CWMS_20"."CWMS_LEVEL"."RETRIEVE_LOCATION_LEVELS_F" (?, ?, ?, ?, ?, ?, ?, ?) }]; ORA-20010: INVALID_OFFICE_ID: "UTC" is not a valid CWMS office id
ORA-06512: at "CWMS_20.CWMS_ERR", line 59
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9482
ORA-01403: no data found
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9479
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9333
ORA-06512: at line 1

	at org.jooq_3.11.2.ORACLE11G.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2380)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:811)
	at org.jooq.impl.AbstractRoutine.executeCallableStatement(AbstractRoutine.java:510)
	at org.jooq.impl.AbstractRoutine.execute(AbstractRoutine.java:391)
	at org.jooq.impl.AbstractRoutine.execute(AbstractRoutine.java:321)
	at usace.cwms.db.jooq.codegen.packages.CWMS_LEVEL_PACKAGE.call_RETRIEVE_LOCATION_LEVELS_F(CWMS_LEVEL_PACKAGE.java:1557)
	at cwms.radar.data.CwmsDataManager.getLocationLevels(CwmsDataManager.java:60)
	at cwms.radar.api.LevelsController.getAll(LevelsController.java:219)
	at io.javalin.apibuilder.CrudFunction$1.invoke$lambda-0(CrudHandler.kt:30)
	at io.javalin.apibuilder.CrudFunctionHandler.handle(CrudHandler.kt)
	at io.javalin.core.security.SecurityUtil.noopAccessManager(SecurityUtil.kt:20)
	at io.javalin.http.JavalinServlet.addHandler$lambda-5(JavalinServlet.kt:113)
	at io.javalin.http.JavalinServlet$service$tryBeforeAndEndpointHandlers$1.invoke(JavalinServlet.kt:44)
	at io.javalin.http.JavalinServlet$service$tryBeforeAndEndpointHandlers$1.invoke(JavalinServlet.kt:39)
	at io.javalin.http.JavalinServlet.service$tryWithExceptionMapper(JavalinServlet.kt:129)
	at io.javalin.http.JavalinServlet.service$tryBeforeAndEndpointHandlers(JavalinServlet.kt:39)
	at io.javalin.http.JavalinServlet.service(JavalinServlet.kt:87)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	at cwms.radar.ApiServlet.service(ApiServlet.java:205)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:540)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:382)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:895)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1722)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.sql.SQLException: ORA-20010: INVALID_OFFICE_ID: "UTC" is not a valid CWMS office id
ORA-06512: at "CWMS_20.CWMS_ERR", line 59
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9482
ORA-01403: no data found
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9479
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9333
ORA-06512: at line 1

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:509)
	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:461)
	at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1104)
	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:550)
	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:268)
	at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:655)
	at oracle.jdbc.driver.T4CCallableStatement.doOall8(T4CCallableStatement.java:265)
	at oracle.jdbc.driver.T4CCallableStatement.doOall8(T4CCallableStatement.java:86)
	at oracle.jdbc.driver.T4CCallableStatement.executeForRows(T4CCallableStatement.java:965)
	at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1205)
	at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3666)
	at oracle.jdbc.driver.T4CCallableStatement.executeInternal(T4CCallableStatement.java:1358)
	at oracle.jdbc.driver.OraclePreparedStatement.execute(OraclePreparedStatement.java:3778)
	at oracle.jdbc.driver.OracleCallableStatement.execute(OracleCallableStatement.java:4251)
	at oracle.jdbc.driver.OraclePreparedStatementWrapper.execute(OraclePreparedStatementWrapper.java:1081)
	at sun.reflect.GeneratedMethodAccessor36.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tomcat.jdbc.pool.StatementFacade$StatementProxy.invoke(StatementFacade.java:118)
	at com.sun.proxy.$Proxy8.execute(Unknown Source)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:209)
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:3416)
	at org.jooq.impl.AbstractRoutine.execute0(AbstractRoutine.java:521)
	at org.jooq.impl.AbstractRoutine.executeCallableStatement(AbstractRoutine.java:481)
	... 35 more
Caused by: Error : 20010, Position : 0, Sql = BEGIN :1 := "CWMS_20"."CWMS_LEVEL"."RETRIEVE_LOCATION_LEVELS_F" (:2 , :3 , :4 , :5 , :6 , :7 , :8 , :9 ) ; END;, OriginalSql = { ? = call "CWMS_20"."CWMS_LEVEL"."RETRIEVE_LOCATION_LEVELS_F" (?, ?, ?, ?, ?, ?, ?, ?) }, Error Msg = ORA-20010: INVALID_OFFICE_ID: "UTC" is not a valid CWMS office id
ORA-06512: at "CWMS_20.CWMS_ERR", line 59
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9482
ORA-01403: no data found
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9479
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9333
ORA-06512: at line 1

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:513)
	... 58 more
```

There seems to be an issue with the start and end dates getting confused with the timezone too:

cURL:
```
curl -X 'GET' \
  'http://localhost:11524/cwms-data/levels?name=AARK.Flow.Inst.0.NWS%20Flood&datum=NAVD88&begin=1997-07-16T19%3A20%3A30Z&end=1997-07-16T19%3A20%3A30Z&timezone=SWT' \
  -H 'accept: */*'
```



```
Nov 18, 2021 4:28:59 PM cwms.radar.ApiServlet lambda$init$9
WARNING: error on request[2780820154355068203]: /cwms-data/levels
org.jooq.exception.DataAccessException: SQL [{ ? = call "CWMS_20"."CWMS_LEVEL"."RETRIEVE_LOCATION_LEVELS_F" (?, ?, ?, ?, ?, ?, ?, ?) }]; ORA-20103: INVALID_TIME_ZONE: The time zone "1997-07-16T19:20:30Z" is not a valid Oracle time zone region
ORA-06512: at "CWMS_20.CWMS_ERR", line 59
ORA-06512: at "CWMS_20.CWMS_UTIL", line 1322
ORA-01403: no data found
ORA-06512: at "CWMS_20.CWMS_UTIL", line 1313
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9559
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9333
ORA-06512: at line 1

	at org.jooq_3.11.2.ORACLE11G.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2380)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:811)
	at org.jooq.impl.AbstractRoutine.executeCallableStatement(AbstractRoutine.java:510)
	at org.jooq.impl.AbstractRoutine.execute(AbstractRoutine.java:391)
	at org.jooq.impl.AbstractRoutine.execute(AbstractRoutine.java:321)
	at usace.cwms.db.jooq.codegen.packages.CWMS_LEVEL_PACKAGE.call_RETRIEVE_LOCATION_LEVELS_F(CWMS_LEVEL_PACKAGE.java:1557)
	at cwms.radar.data.CwmsDataManager.getLocationLevels(CwmsDataManager.java:60)
	at cwms.radar.api.LevelsController.getAll(LevelsController.java:219)
	at io.javalin.apibuilder.CrudFunction$1.invoke$lambda-0(CrudHandler.kt:30)
	at io.javalin.apibuilder.CrudFunctionHandler.handle(CrudHandler.kt)
	at io.javalin.core.security.SecurityUtil.noopAccessManager(SecurityUtil.kt:20)
	at io.javalin.http.JavalinServlet.addHandler$lambda-5(JavalinServlet.kt:113)
	at io.javalin.http.JavalinServlet$service$tryBeforeAndEndpointHandlers$1.invoke(JavalinServlet.kt:44)
	at io.javalin.http.JavalinServlet$service$tryBeforeAndEndpointHandlers$1.invoke(JavalinServlet.kt:39)
	at io.javalin.http.JavalinServlet.service$tryWithExceptionMapper(JavalinServlet.kt:129)
	at io.javalin.http.JavalinServlet.service$tryBeforeAndEndpointHandlers(JavalinServlet.kt:39)
	at io.javalin.http.JavalinServlet.service(JavalinServlet.kt:87)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	at cwms.radar.ApiServlet.service(ApiServlet.java:205)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:540)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:382)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:895)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1722)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.sql.SQLException: ORA-20103: INVALID_TIME_ZONE: The time zone "1997-07-16T19:20:30Z" is not a valid Oracle time zone region
ORA-06512: at "CWMS_20.CWMS_ERR", line 59
ORA-06512: at "CWMS_20.CWMS_UTIL", line 1322
ORA-01403: no data found
ORA-06512: at "CWMS_20.CWMS_UTIL", line 1313
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9559
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9333
ORA-06512: at line 1

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:509)
	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:461)
	at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1104)
	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:550)
	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:268)
	at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:655)
	at oracle.jdbc.driver.T4CCallableStatement.doOall8(T4CCallableStatement.java:265)
	at oracle.jdbc.driver.T4CCallableStatement.doOall8(T4CCallableStatement.java:86)
	at oracle.jdbc.driver.T4CCallableStatement.executeForRows(T4CCallableStatement.java:965)
	at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1205)
	at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3666)
	at oracle.jdbc.driver.T4CCallableStatement.executeInternal(T4CCallableStatement.java:1358)
	at oracle.jdbc.driver.OraclePreparedStatement.execute(OraclePreparedStatement.java:3778)
	at oracle.jdbc.driver.OracleCallableStatement.execute(OracleCallableStatement.java:4251)
	at oracle.jdbc.driver.OraclePreparedStatementWrapper.execute(OraclePreparedStatementWrapper.java:1081)
	at sun.reflect.GeneratedMethodAccessor36.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tomcat.jdbc.pool.StatementFacade$StatementProxy.invoke(StatementFacade.java:118)
	at com.sun.proxy.$Proxy8.execute(Unknown Source)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:209)
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:3416)
	at org.jooq.impl.AbstractRoutine.execute0(AbstractRoutine.java:521)
	at org.jooq.impl.AbstractRoutine.executeCallableStatement(AbstractRoutine.java:481)
	... 35 more
Caused by: Error : 20103, Position : 0, Sql = BEGIN :1 := "CWMS_20"."CWMS_LEVEL"."RETRIEVE_LOCATION_LEVELS_F" (:2 , :3 , :4 , :5 , :6 , :7 , :8 , :9 ) ; END;, OriginalSql = { ? = call "CWMS_20"."CWMS_LEVEL"."RETRIEVE_LOCATION_LEVELS_F" (?, ?, ?, ?, ?, ?, ?, ?) }, Error Msg = ORA-20103: INVALID_TIME_ZONE: The time zone "1997-07-16T19:20:30Z" is not a valid Oracle time zone region
ORA-06512: at "CWMS_20.CWMS_ERR", line 59
ORA-06512: at "CWMS_20.CWMS_UTIL", line 1322
ORA-01403: no data found
ORA-06512: at "CWMS_20.CWMS_UTIL", line 1313
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9559
ORA-06512: at "CWMS_20.CWMS_LEVEL", line 9333
ORA-06512: at line 1

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:513)
	... 58 more

```

--- a/cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java
+++ b/cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java
@@ -2,23 +2,17 @@ package cwms.radar.api;
 
 import java.io.IOException;
 import java.math.BigDecimal;
-import java.sql.SQLException;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
-
 import javax.servlet.http.HttpServletResponse;
 
 import com.codahale.metrics.Histogram;
 import com.codahale.metrics.Meter;
 import com.codahale.metrics.MetricRegistry;
-import static com.codahale.metrics.MetricRegistry.*;
-import static cwms.radar.data.dao.JooqDao.getDslContext;
-
 import com.codahale.metrics.Timer;
-
 import com.fasterxml.jackson.databind.BeanDescription;
 import com.fasterxml.jackson.databind.JavaType;
 import com.fasterxml.jackson.databind.JsonNode;
@@ -29,8 +23,8 @@ import com.fasterxml.jackson.dataformat.xml.XmlMapper;
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import cwms.radar.api.enums.UnitSystem;
 import cwms.radar.api.errors.RadarError;
-import cwms.radar.data.CwmsDataManager;
-import cwms.radar.data.dao.*;
+import cwms.radar.data.dao.LocationLevelsDao;
+import cwms.radar.data.dao.LocationLevelsDaoImpl;
 import cwms.radar.data.dto.LocationLevel;
 import cwms.radar.data.dto.SeasonalValueBean;
 import cwms.radar.formatters.ContentType;
@@ -39,9 +33,17 @@ import cwms.radar.formatters.FormattingException;
 import cwms.radar.formatters.xml.adapters.ZonedDateTimeAdapter;
 import io.javalin.apibuilder.CrudHandler;
 import io.javalin.http.Context;
-import io.javalin.plugin.openapi.annotations.*;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
 import org.jetbrains.annotations.NotNull;
 import org.jooq.DSLContext;
+
+import static com.codahale.metrics.MetricRegistry.name;
+import static cwms.radar.data.dao.JooqDao.getDslContext;
 
 public class LevelsController implements CrudHandler {
     private static final Logger logger = Logger.getLogger(LevelsController.class.getName());
@@ -191,8 +193,10 @@ public class LevelsController implements CrudHandler {
         getAllRequests.mark();
         try (
             final Timer.Context time_context = getAllRequestsTime.time();
-            CwmsDataManager cdm = new CwmsDataManager(ctx);
+            DSLContext dsl = getDslContext(ctx);
         ) {
+            LocationLevelsDao levelsDao = getLevelsDao(dsl);
+
             String format = ctx.queryParamAsClass("format",String.class).getOrDefault("json");
             String names = ctx.queryParam("name");
             String office = ctx.queryParam("office");
@@ -216,14 +220,10 @@ public class LevelsController implements CrudHandler {
                 }
             }
 
-            String results = cdm.getLocationLevels(format,names,office,unit,datum,begin,end,timezone);
+            String results = levelsDao.getLocationLevels(format,names,office,unit,datum,begin,end,timezone);
             ctx.status(HttpServletResponse.SC_OK);
             ctx.result(results);
             requestResultSize.update(results.length());
-        } catch (SQLException ex) {
-            RadarError re = new RadarError("Internal Error");
-            logger.log(Level.SEVERE, re.toString(), ex);
-            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(re);
         }
     }
 

--- a/cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java
+++ b/cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java
@@ -10,7 +10,6 @@ import java.util.logging.Logger;
 import javax.servlet.http.HttpServletResponse;
 
 import com.codahale.metrics.Histogram;
-import com.codahale.metrics.Meter;
 import com.codahale.metrics.MetricRegistry;
 import com.codahale.metrics.Timer;
 import com.fasterxml.jackson.databind.BeanDescription;
@@ -48,32 +47,19 @@ import static cwms.radar.data.dao.JooqDao.getDslContext;
 public class LevelsController implements CrudHandler {
     private static final Logger logger = Logger.getLogger(LevelsController.class.getName());
     private final MetricRegistry metrics;
-    private final Meter getAllRequests;
-    private final Timer getAllRequestsTime;
-    private final Meter getOneRequest;
-    private final Timer getOneRequestTime;
+
     private final Histogram requestResultSize;
-    private final Meter createRequest;
-    private final Timer createRequestTime;
-    private final Meter deleteRequest;
-    private final Timer deleteRequestTime;
-    private final Meter updateRequest;
-    private final Timer updateRequestTime;
+
 
     public LevelsController(MetricRegistry metrics){
         this.metrics=metrics;
-        String className = this.getClass().getName();
-        getAllRequests = this.metrics.meter(name(className,"getAll","count"));
-        getAllRequestsTime = this.metrics.timer(name(className,"getAll","time"));
-        getOneRequest = this.metrics.meter(name(className,"getOne","count"));
-        getOneRequestTime = this.metrics.timer(name(className,"getOne","time"));
-        createRequest = this.metrics.meter(name(className,"create","count"));
-        createRequestTime = this.metrics.timer(name(className,"create","time"));
-        deleteRequest = this.metrics.meter(name(className,"delete","count"));
-        deleteRequestTime = this.metrics.timer(name(className,"delete","time"));
-        updateRequest = this.metrics.meter(name(className,"update","count"));
-        updateRequestTime = this.metrics.timer(name(className,"update","time"));
-        requestResultSize = this.metrics.histogram((name(className,"results","size")));
+
+        requestResultSize = this.metrics.histogram((name(this.getClass().getName(),"results","size")));
+    }
+
+    private Timer.Context markAndTime(String subject)
+    {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
     }
 
     @OpenApi(
@@ -94,8 +80,8 @@ public class LevelsController implements CrudHandler {
     @Override
     public void create(@NotNull Context ctx)
     {
-        createRequest.mark();
-        try(final Timer.Context timeContext = createRequestTime.time();
+
+        try(final Timer.Context timeContext = markAndTime("create");
             DSLContext dsl = getDslContext(ctx))
         {
             LocationLevelsDao levelsDao = getLevelsDao(dsl);
@@ -150,8 +136,8 @@ public class LevelsController implements CrudHandler {
     @Override
     public void delete(@NotNull Context ctx, String id)
     {
-        deleteRequest.mark();
-        try(final Timer.Context timeContext = deleteRequestTime.time();
+
+        try(final Timer.Context timeContext = markAndTime("delete");
             DSLContext dsl = getDslContext(ctx))
         {
             String office = ctx.queryParam("office");
@@ -173,14 +159,14 @@ public class LevelsController implements CrudHandler {
 
     @OpenApi(
         queryParams = {
-            @OpenApiParam(name="name", required=false, description="Specifies the name(s) of the location level(s) whose data is to be included in the response. Uses * for all."),
-            @OpenApiParam(name="office", required=false, description="Specifies the owning office of the location level(s) whose data is to be included in the response. If this field is not specified, matching location level information from all offices shall be returned."),
-            @OpenApiParam(name="unit", required=false, description="Specifies the unit or unit system of the response. Valid values for the unit field are:\r\n 1. EN.   Specifies English unit system.  Location level values will be in the default English units for their parameters.\r\n2. SI.   Specifies the SI unit system.  Location level values will be in the default SI units for their parameters.\r\n3. Other. Any unit returned in the response to the units URI request that is appropriate for the requested parameters."),
-            @OpenApiParam(name="datum", required=false, description="Specifies the elevation datum of the response. This field affects only elevation location levels. Valid values for this field are:\r\n1. NAVD88.  The elevation values will in the specified or default units above the NAVD-88 datum.\r\n2. NGVD29.  The elevation values will be in the specified or default units above the NGVD-29 datum."),
-            @OpenApiParam(name="begin", required=false, description="Specifies the start of the time window for data to be included in the response. If this field is not specified, any required time window begins 24 hours prior to the specified or default end time."),
-            @OpenApiParam(name="end", required=false, description="Specifies the end of the time window for data to be included in the response. If this field is not specified, any required time window ends at the current time"),
-            @OpenApiParam(name="timezone", required=false, description="Specifies the time zone of the values of the begin and end fields (unless otherwise specified), as well as the time zone of any times in the response. If this field is not specified, the default time zone of UTC shall be used."),
-            @OpenApiParam(name="format", required=false, description="Specifies the encoding format of the response. Valid values for the format field for this URI are:\r\n1.    tab\r\n2.    csv\r\n3.    xml\r\n4.  wml2 (only if name field is specified)\r\n5.    json (default)")
+            @OpenApiParam(name="name", description="Specifies the name(s) of the location level(s) whose data is to be included in the response. Uses * for all."),
+            @OpenApiParam(name="office", description="Specifies the owning office of the location level(s) whose data is to be included in the response. If this field is not specified, matching location level information from all offices shall be returned."),
+            @OpenApiParam(name="unit", description="Specifies the unit or unit system of the response. Valid values for the unit field are:\r\n 1. EN.   Specifies English unit system.  Location level values will be in the default English units for their parameters.\r\n2. SI.   Specifies the SI unit system.  Location level values will be in the default SI units for their parameters.\r\n3. Other. Any unit returned in the response to the units URI request that is appropriate for the requested parameters."),
+            @OpenApiParam(name="datum", description="Specifies the elevation datum of the response. This field affects only elevation location levels. Valid values for this field are:\r\n1. NAVD88.  The elevation values will in the specified or default units above the NAVD-88 datum.\r\n2. NGVD29.  The elevation values will be in the specified or default units above the NGVD-29 datum."),
+            @OpenApiParam(name="begin", description="Specifies the start of the time window for data to be included in the response. If this field is not specified, any required time window begins 24 hours prior to the specified or default end time."),
+            @OpenApiParam(name="end", description="Specifies the end of the time window for data to be included in the response. If this field is not specified, any required time window ends at the current time"),
+            @OpenApiParam(name="timezone", description="Specifies the time zone of the values of the begin and end fields (unless otherwise specified), as well as the time zone of any times in the response. If this field is not specified, the default time zone of UTC shall be used."),
+            @OpenApiParam(name="format", description="Specifies the encoding format of the response. Valid values for the format field for this URI are:\r\n1.    tab\r\n2.    csv\r\n3.    xml\r\n4.  wml2 (only if name field is specified)\r\n5.    json (default)")
         },
         responses = {
             @OpenApiResponse(status="200" )
@@ -190,10 +176,10 @@ public class LevelsController implements CrudHandler {
     @Override
     public void getAll(Context ctx)
     {
-        getAllRequests.mark();
+
         try (
-            final Timer.Context time_context = getAllRequestsTime.time();
-            DSLContext dsl = getDslContext(ctx);
+            final Timer.Context timeContext = markAndTime("getAll");
+            DSLContext dsl = getDslContext(ctx)
         ) {
             LocationLevelsDao levelsDao = getLevelsDao(dsl);
 
@@ -253,8 +239,8 @@ public class LevelsController implements CrudHandler {
     @Override
     public void update(@NotNull Context ctx, String id)
     {
-        updateRequest.mark();
-        try(final Timer.Context timeContext = updateRequestTime.time();
+
+        try(final Timer.Context timeContext = markAndTime("update");
             DSLContext dsl = getDslContext(ctx))
         {
             LocationLevelsDao levelsDao = getLevelsDao(dsl);

--- a/cwms_radar_api/src/main/java/cwms/radar/data/CwmsDataManager.java
+++ b/cwms_radar_api/src/main/java/cwms/radar/data/CwmsDataManager.java
@@ -58,7 +58,7 @@ public class CwmsDataManager implements AutoCloseable {
 	public String getLocationLevels(String format, String names, String office, String unit, String datum, String begin,
 			String end, String timezone) {
         return CWMS_LEVEL_PACKAGE.call_RETRIEVE_LOCATION_LEVELS_F(dsl.configuration(),
-                names, format, office,unit,datum, begin, end, timezone);
+                names, format, unit, datum, begin, end, timezone, office);
     }
 
 

--- a/cwms_radar_api/src/main/java/cwms/radar/data/CwmsDataManager.java
+++ b/cwms_radar_api/src/main/java/cwms/radar/data/CwmsDataManager.java
@@ -11,7 +11,6 @@ import org.jooq.impl.DSL;
 
 import usace.cwms.db.jooq.codegen.packages.CWMS_CAT_PACKAGE;
 import usace.cwms.db.jooq.codegen.packages.CWMS_ENV_PACKAGE;
-import usace.cwms.db.jooq.codegen.packages.CWMS_LEVEL_PACKAGE;
 
 
 public class CwmsDataManager implements AutoCloseable {
@@ -53,12 +52,6 @@ public class CwmsDataManager implements AutoCloseable {
 
 	public String getTimeZones(String format) {
             return CWMS_CAT_PACKAGE.call_RETRIEVE_TIME_ZONES_F(dsl.configuration(), format);
-    }
-
-	public String getLocationLevels(String format, String names, String office, String unit, String datum, String begin,
-			String end, String timezone) {
-        return CWMS_LEVEL_PACKAGE.call_RETRIEVE_LOCATION_LEVELS_F(dsl.configuration(),
-                names, format, unit, datum, begin, end, timezone, office);
     }
 
 

--- a/cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationLevelsDao.java
+++ b/cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationLevelsDao.java
@@ -12,4 +12,6 @@ public interface LocationLevelsDao
     void storeLocationLevel(LocationLevel level, ZoneId zoneId) throws IOException;
     void renameLocationLevel(String oldLocationLevelName, LocationLevel renamedLocationLevel) throws IOException;
     LocationLevel retrieveLocationLevel(String locationLevelName, String unitSystem, ZonedDateTime effectiveDate, String officeId) throws IOException;
+    String getLocationLevels(String format, String names, String office, String unit, String datum, String begin,
+                             String end, String timezone);
 }

--- a/cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationLevelsDaoImpl.java
+++ b/cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationLevelsDaoImpl.java
@@ -9,6 +9,7 @@ import org.jooq.exception.DataAccessException;
 import usace.cwms.db.dao.ifc.level.CwmsDbLevel;
 import usace.cwms.db.dao.ifc.level.LocationLevelPojo;
 import usace.cwms.db.dao.util.services.CwmsDbServiceLookup;
+import usace.cwms.db.jooq.codegen.packages.CWMS_LEVEL_PACKAGE;
 
 import java.io.IOException;
 import java.math.BigInteger;
@@ -29,6 +30,14 @@ public class LocationLevelsDaoImpl extends JooqDao<LocationLevel> implements Loc
     public LocationLevelsDaoImpl(DSLContext dsl)
     {
         super(dsl);
+    }
+
+    // This is the legacy method that is used by the old API.
+    @Override
+    public String getLocationLevels(String format, String names, String office, String unit, String datum, String begin,
+                                    String end, String timezone) {
+        return CWMS_LEVEL_PACKAGE.call_RETRIEVE_LOCATION_LEVELS_F(dsl.configuration(),
+                names, format, unit, datum, begin, end, timezone, office);
     }
 
     @Override

